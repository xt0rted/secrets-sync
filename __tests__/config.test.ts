import { fileURLToPath } from "node:url";

import {
  loadConfig,
  readConfigValue,
} from "../src/config";

describe("config", () => {
  describe("loadConfig", () => {
    test("should return a config", async () => {
      const configFile = fileURLToPath(new URL("test-config.yml", import.meta.url));

      expect(await loadConfig(configFile)).toMatchSnapshot();
    });
  });

  describe("readConfigValue", () => {
    afterEach(() => {
      delete process.env["INPUT_VALUE"];
    });

    test("should return undefined if undefined is provided", () => {
      // eslint-disable-next-line unicorn/no-useless-undefined
      expect(readConfigValue(undefined)).toBe(undefined);
    });

    test("should return null if null is provided", () => {
      // eslint-disable-next-line unicorn/no-null
      expect(readConfigValue(null!)).toBe(null);
    });

    test("should return the value if it's not an env variable", () => {
      expect(readConfigValue("TEST_VALUE")).toEqual("TEST_VALUE");
    });

    test("should return the value if it's an env variable", () => {
      process.env["TEST_VALUE"] = "test";

      expect(readConfigValue("env/TEST_VALUE")).toEqual("test");
    });
  });
});
