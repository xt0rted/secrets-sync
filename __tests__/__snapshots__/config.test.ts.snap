// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config loadConfig should return a config 1`] = `
Object {
  "defaults": Object {
    "actions": true,
    "dependabot": true,
  },
  "secrets": Array [
    Object {
      "name": "ENV_VAR1",
      "repos": Array [
        "test/repo",
      ],
      "value": "abc123",
    },
    Object {
      "name": "ENV_VAR2",
      "repos": Array [
        "test/repo",
      ],
      "value": "env/ENV_VAR2",
    },
    Object {
      "name": "ENV_VAR3",
      "repos": Array [
        "test/repo",
      ],
    },
    Object {
      "actions": false,
      "dependabot": false,
      "name": "ENV_VAR4",
      "repos": Array [
        "test/repo",
      ],
      "value": "123abc",
    },
  ],
}
`;
